lib/3scale_toolbox/commands/import_command/openapi/update_service_proxy_step.rb

```diff
@@ -0,0 +1,67 @@
+module ThreeScaleToolbox
+  module Commands
+    module ImportCommand
+      module OpenAPI
+        class UpdateServiceProxyStep
+          include Step
+
+          ##
+          # Updates Proxy config
+          def call
+            # setting required attrs, operation is idempotent
+            proxy_settings = {}
+
+            add_api_backend_settings(proxy_settings)
+            add_security_proxy_settings(proxy_settings)
+
+            return unless proxy_settings.size.positive?
+
+            service.update_proxy proxy_settings
+            puts 'Service proxy updated'
+          end
+
+          private
+
+          def add_api_backend_settings(settings)
+            scheme = api_spec.schemes.first || 'https'
+            host = api_spec.host
+
+            settings[:api_backend] = "#{scheme}://#{host}"
+            nil
+          end
+
+          def add_security_proxy_settings(settings)
+            # nothing to add on proxy settings when no security required in openapi
+            return if security.nil?
+
+            case security.type
+            when 'oauth2'
+              settings[:credentials_location] = 'headers'
+              settings[:oidc_issuer_endpoint] = oidc_issuer_endpoint unless oidc_issuer_endpoint.nil?
+            when 'apiKey'
+              settings[:credentials_location] = credentials_location
+              settings[:auth_user_key] = security.name
+            else
+              raise ThreeScaleToolbox::Error, "Unexpected security scheme type #{security.type}"
+            end
+
+            nil
+          end
+
+          def credentials_location
+            return nil if security.nil?
+
+            case security.in_f
+            when 'query'
+              'query'
+            when 'header'
+              'headers'
+            else
+              raise ThreeScaleToolbox::Error, "Unexpected security in_f field #{security.in_f}"
```

Comment:
This is a potentially untested branch with a risk of having a typo in the interpolation. Using variables would prevent that `case (in_f = security.in_f)`.

Reply:
Adding test for that branch

Reply:
My point wasn't really about this one branch, but general about not calling a method in a less used branch and rather using a variable.